### ISSUE TITLE
[invalid-parameter-default] on ellipsis in `TYPE_CHECKING` section

### ISSUE THREAD

**Issue by robsdedude** [astral-sh/ty#2534]
### Summary

https://peps.python.org/pep-0484/#arbitrary-argument-lists-and-default-argument-values reads

> In stubs it may be useful to declare an argument as having a default without specifying the actual default value. For example:
>
> ```python
> def foo(x: AnyStr, y: AnyStr = ...) -> AnyStr: ...
> ```

I argue, in the same vein as https://github.com/astral-sh/ty/issues/339, that `if TYPE_CHECKING` sections are very stub-file-like. Therefore, I suggest to allow assigning `...` (ellipsis) as a parameter default value. At least `mypy` is also fine with this.

Admittedly, mypy just allows ellipsis to be a parameter default everywhere. I'm not sure that's a good idea, really. mypy gist: https://gist.github.com/mypy-play/41532e8350478fd1a0b4fa6ee764171b

**Comment by AlexWaygood** [comment]
yeah fair enough! This is easy to fix
